### DISCUSSION TITLE
adding warehouses to existing db?

### DISCUSSION THREAD

**Discussion by SQLAmanda** [TPC-Council/HammerDB#219] [Q&A] [answered]
I'd like to add more data for my tcpc-c test. It seemed to take forever to load 200 warehouses. I can't remember the user ratio, but eventually things timed out and it never built the indexes or stored procedures.
I'm running this to test an average cloud setup, i've got maybe 8GB of data.
1 - can I add more warehouses to an existing db, if so, how?
2- what's the best ratio of users to warehouses to get the most data? I'm running from an 8CPU VM (in the same cloud, etc.)

The db has 2 vcpu. Is tiny.

**Comment by sm-shaw** [comment, marked as answer]
Yes it is possible and would involve saving and then modifying a custom build script.  However, it would be best to address the core issue as to why it is taking so long to load the schema instead. As a guide using MySQL we see loading 1000 warehouses takes 10 minutes. It will load in parallel depending on how many virtual users you have created to do the build. 
So the key question is why is it taking so long to load? - in the cloud addressing the network round trip is important, if however as you say the client is in the same cloud as the database then this shouldn't necessarily be an issue (but please verify this). If you are sure it's not the network round trip then another key area is often the redo/transaction log/wal performance by default the load is done using inserts so all of the workload will be logged, therefore the log performance is important.  
An alternative intentionally designed for the cloud if you are having difficulty loading is to use the datagen feature https://www.hammerdb.com/docs/ch13.html this allows you to create flatfiles and bulk load them bypassing the bottlenecks with an insert based approach but needs 2X the amount of data as you create both the flat files and then load them. 
For the warehouses to create please use the following reference:
https://www.hammerdb.com/docs/ch03s07.html
https://www.hammerdb.com/blog/uncategorized/how-many-warehouses-for-the-hammerdb-tpc-c-test/

**Reply by SQLAmanda** [reply]
i'm still unsure why it takes so long. I suspect netowking issues. Thank you for the suggestion on data-gen.

**Reply by SQLAmanda** [reply]
If i want to load 1000 warehouses, how many users should i use to load them?

**Reply by sm-shaw** [reply]
The main factor is the number of cores/threads running on the client ie where HammerDB is running. The load is both creating and inserting the data multiple rows at a time so a lot of CPU will be used in creating the data, this is intentional as you do not need flat files as an intermediary to create and load the data.  Another restriction is the database engine itself as some databases cope better with inserting into the same table by a large number of users better than others, this can be found by trial and error.
